### DISCUSSION TITLE
Will there be other field other than id?

### DISCUSSION THREAD

**Discussion by linonetwo** [comunica/comunica#445] [Q&A] [answered]
#### Issue type:

- :question: Question

____
#### Question:

Can I just use `data.get('?s').id`?

Where can I found a document, I've searched through http://rdf.js.org/#source-interface but still don't know where to look at...

**Comment by rubensworks** [comment, marked as answer]
All result bindings are RDFJS terms according to this spec: http://rdf.js.org/data-model-spec/

The raw value will typically be retrievable using `.value`. Definitely do not use `.id`, as this is an internal field of N3.js, which will not be available if you use different parsers/stores.

**Comment by linonetwo** [comment]
Thanks for the explanation.

I tried to log it out:

```js
result.bindingsStream.on('data', data => {
  console.log(data.get('?s'), data.get('?p'), data.get('?o'));
});
```
got
```
Object { id: "null" }
Object { id: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" }
Object { id: "http://www.w3.org/ns/ldp#BasicContainer" }

Object { id: "null" }
Object { id: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" }
Object { id: "http://www.w3.org/ns/ldp#Container" }
```
So I was confused.

**Comment by linonetwo** [comment]
So what returned from `data.get('?s')` is a http://rdf.js.org/data-model-spec/#variable-interface right?

**Comment by rubensworks** [comment]
No, this will actually never be Variable. This will either be NamedNode, Literal or BlankNode, depending on the types of RDF terms that you are querying.
